### PR TITLE
Use official golang image to build operator

### PR DESCRIPTION
**Pull request by janisz**
## Description

Currently we are using stream8 and install dependencies for every build. This PR replaces it with official golang image so we have pinned golang version but still get latest updates to it.

This should prevent issues like this: https://app.circleci.com/pipelines/github/stackrox/stackrox/8929/workflows/614b3394-3a73-4ed9-8426-f1aac1f01d28/jobs/396277?invite=true#step-109-199

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [stackrox/openshift-docs](https://github.com/stackrox/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

CI

### CODE REVIEW

**Review by gavin-stackrox — APPROVED** [review]
LGTM if it can pass CI which may not be possible today.

**Comment by janisz** [comment]
Rebased, let's see if tests pass

File: operator/Dockerfile [hunk 1]
```diff
@@ -20,10 +16,6 @@ COPY go.sum go.sum
 # Add GitHub hosts as known ones so that git does not complain.
 RUN mkdir -p -m 0700 ~/.ssh && ssh-keyscan github.com >> ~/.ssh/known_hosts
```
Comment (porridge) [inline operator/Dockerfile:17]: I wonder whether we even use git+ssh now that the `insteadOf` below is gone and the repo is public...
Not in scope of this PR, but neither is the below change ;-)

**Review by porridge — APPROVED** [review]

File: operator/Dockerfile (hunk 1 above)
Comment (janisz) [reply operator/Dockerfile]: Removed it

**Pull request merged**